internal/cmd/ssh.go

```diff
@@ -0,0 +1,388 @@
+	if err != nil {
+		return err
+	}
+
+	if err := writeDestinationSSHConfig(infraSSHDir, destination, user); err != nil {
+		return fmt.Errorf("write infra ssh config: %w", err)
+	}
+	return nil
+}
+
+func provisionSSHKey(ctx context.Context, cli *CLI, client *api.Client, infraSSHDir string) error {
+	keyFilename := filepath.Join(infraSSHDir, "key")
+
+	// TODO: check expiration
+	// TODO: check the key exists in the API
+	if fileExists(keyFilename) {
+		return nil
+	}
+
+	fmt.Fprintf(cli.Stderr, "Creating a new RSA 4096 bit key pair in %v\n", keyFilename)
+
+	priv, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return fmt.Errorf("generate key pair: %w", err)
+	}
+
+	fh, err := os.OpenFile(keyFilename, os.O_RDWR|os.O_TRUNC|os.O_CREATE, 0600)
+	if err != nil {
+		return err
+	}
+	block := &pem.Block{
+		Type:  "RSA PRIVATE KEY",
+		Bytes: x509.MarshalPKCS1PrivateKey(priv),
+	}
+	if err := pem.Encode(fh, block); err != nil {
+		return err
+	}
+	if err := fh.Close(); err != nil {
+		return err
+	}
+
+	sshPubKey, err := ssh.NewPublicKey(&priv.PublicKey)
+	if err != nil {
+		return err
+	}
+
+	pubKeyBytes := ssh.MarshalAuthorizedKey(sshPubKey)
+	if err := os.WriteFile(keyFilename+".pub", pubKeyBytes, 0600); err != nil {
+		return err
+	}
+
+	hostname, _ := os.Hostname()
+	_, err = client.AddUserPublicKey(ctx, &api.AddUserPublicKeyRequest{
+		Name:      hostname,
+		PublicKey: string(pubKeyBytes),
+	})
+	return err
+}
+
+func fileExists(filename string) bool {
+	_, err := os.Stat(filename)
+	return err == nil
+}
+
+func updateUserSSHConfig(cli *CLI) error {
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		return fmt.Errorf("user home directory: %w", err)
+	}
+
+	userSSHDir := filepath.Join(homeDir, ".ssh")
+	filename := filepath.Join(userSSHDir, "config")
+
+	var original []byte
+	fh, err := os.Open(filename)
+	switch {
+	case errors.Is(err, fs.ErrNotExist):
+		// file is missing, we'll create it later
+		_ = os.MkdirAll(userSSHDir, 0700)
+	case err != nil:
+		return fmt.Errorf("open ssh config: %w", err)
+	default:
+		defer fh.Close() // closing the read only file
+
+		if hasInfraMatchLine(fh) {
+			return nil
+		}
+
+		_, err = fh.Seek(0, io.SeekStart)
+		if err != nil {
+			return fmt.Errorf("seek: %w", err)
+		}
+
+		original, err = io.ReadAll(fh)
+		if err != nil {
+			return fmt.Errorf("read file: %w", err)
+		}
+
+		_ = fh.Close() // closing the read only file
+	}
+
+	tmp, err := os.CreateTemp(userSSHDir, "config-")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(tmp.Name())
+
+	if _, err := tmp.Write(original); err != nil {
+		return err
+	}
+
+	if _, err := tmp.Write([]byte(infraUserSSHConfig)); err != nil {
+		return err
+	}
+	if err := tmp.Sync(); err != nil {
+		return err
+	}
+	if err := tmp.Close(); err != nil {
+		return err
+	}
+
+	if err := os.Rename(tmp.Name(), filename); err != nil {
+		return err
+	}
+	cli.Output(`
+Your SSH config at %v has been created or updated to use 'infra ssh hosts' for
```

Comment:
```suggestion
Your SSH config at %v has been updated to connect to Infra SSH destinations.
```

My suggestion is we don't talk about `infra ssh hosts` since it's a hidden command. At the same time, we want to make sure users don't feel Infra is too invasive.